File: app/src/main/java/com/akay/fitnass/viewmodel/BaseViewModel.java
package com.akay.fitnass.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.akay.fitnass.App;
import com.akay.fitnass.data.RunsRepository;
import com.akay.fitnass.data.model.ActiveRuns;

public class BaseViewModel extends ViewModel {
    private final RunsRepository mRepository;

    BaseViewModel() {
        mRepository = App.getComponent().getRepository();
    }

    protected RunsRepository getRepo() {
        return mRepository;
    }

    public LiveData<ActiveRuns> getLiveActiveRuns() {
        return getRepo().getLiveActiveRuns();
    }
}

File: app/src/main/java/com/akay/fitnass/viewmodel/BaseViewModel.kt
package com.akay.fitnass.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

import com.akay.fitnass.App
import com.akay.fitnass.data.RunsRepository
import com.akay.fitnass.data.model.ActiveRuns

open class BaseViewModel : ViewModel() {
    protected val repo: RunsRepository = App.component.repository

    fun getLiveActiveRuns(): LiveData<ActiveRuns> = repo.getLiveActiveRuns()

}
